raphtory/src/algorithms/bipartite/max_weight_matching.rs

```diff
@@ -0,0 +1,1391 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// Needed to pass shared state between functions
+// closures don't work because of recurssion
+#![allow(clippy::too_many_arguments)]
```

Comment:
could have put all the state into a struct and made those functions methods instead to avoid having to pass around all the arguments?